src/index.ts

```diff
@@ -93,22 +149,27 @@ class DebuggerHandler<
   async update<W extends ConsolePanel | NotebookPanel | FileEditor>(
     debug: IDebugger,
     widget: W
-  ) {
+  ): Promise<void> {
     const debuggingEnabled = await debug.requestDebuggingEnabled();
     if (!debug.model || this.handlers[widget.id] || !debuggingEnabled) {
+      if (!debug.session.debuggedClients.has(debug.session.client.name)) {
+        return debug.stop();
```

Comment:
This means that `debug.stop()` is called for clients that do not support debugging, which brings back #268 (fixed in #275)

Reply:
This is visible in the notebook server logs:

```
[IPKernelApp] ERROR | UNKNOWN CONTROL MESSAGE TYPE: 'debug_request'
```